mcs/class/Makefile

```diff
@@ -413,24 +412,37 @@ lib/$(monolite_dir)/Facades:
 
 $(monolite_files): | lib/$(monolite_dir)
 $(monolite_files): | lib/$(monolite_dir)/Facades
-$(monolite_files): lib/$(monolite_dir)/%: lib/build/%
+$(monolite_files): lib/$(monolite_dir)/%: lib/build-$(HOST_PLATFORM)/%
 	cp -p $< $@
 
 lib/$(monolite_dir)/mcs.exe:
 	$(MAKE) -C ../mcs PROFILE=build
-	cp -p lib/build/mcs.exe lib/$(monolite_dir)
+	cp -p lib/build-$(HOST_PLATFORM)/mcs.exe lib/$(monolite_dir)
 
-$(build_files:%=lib/build/%):
+$(build_files:%=lib/build-$(HOST_PLATFORM)/%):
 	cd $(topdir) && $(MAKE) profile-do--build--all NO_DIR_CHECK=1 SKIP_AOT=1
 
 dist-monolite: $(monolite_files) lib/$(monolite_dir)/mcs.exe
 
+dist-monolite-all-platforms:
```

Comment:
So this is a bit confusing, but let me explain:

The `package-monolite-latest` target is _not_ used when we're creating a tarball. It is only used as part of our CI to create a new monolite that we put on the download server. I have modified our CI config to call `package-monolite-latest` with different HOST_PLATFORMS set.

When creating a tarball on the other hand, we need to now include a monolite for all the supported platforms. The tarball is created by calling `make dist` which eventually calls into `dist-monolite-all-platforms` which builds the different monolites. They are then included in the tarball because of the entry in DISTFILES.

I do think adding a `package-monolite-latest-all` target is a good idea since we can then move those configs from CI into the Makefile. I'll add that. It doesn't change anything about `dist-monolite-all-platforms` though since those are not related :)